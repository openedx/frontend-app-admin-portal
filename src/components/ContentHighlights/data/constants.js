/* eslint-disable import/no-extraneous-dependencies */
import parse from 'html-react-parser';
import sanitizeHTML from 'sanitize-html';
/* eslint-enable import/no-extraneous-dependencies */

// Sanitizes HTML and parses the string as HTML
export const sanitizeAndParseHTML = (htmlString) => {
  const sanitizedHTML = sanitizeHTML(htmlString);
  return parse(sanitizedHTML);
};

/* START LOCAL TESTING CONSTANTS */
// Set to false before pushing PR!! otherwise set to true to enable local testing of ContentHighlights components
// Test will fail as additional check to ensure this is set to false before pushing PR
export const TEST_FLAG = false;
// Test entepriseId for Content Highlights to display card selections and confirmation
export const testEnterpriseId = 'f23ccd7d-fbbb-411a-824e-c2861942aac0';
// function that passes through enterpriseId if TEST_FLAG is false, otherwise returns local testing enterpriseId
export const ENABLE_TESTING = (enterpriseId, enableTest = TEST_FLAG) => {
  if (enableTest) {
    return testEnterpriseId;
  }
  return enterpriseId;
};
/* END LOCAL TESTING CONSTANTS */

// Default Card Grid columnSizes
export const HIGHLIGHTS_CARD_GRID_COLUMN_SIZES = {
  xs: 12,
  md: 6,
  lg: 4,
  xl: 3,
};

// Tab titles
export const TAB_TITLES = {
  highlights: 'Highlights',
  catalogVisibility: 'Catalog Visibility',
};

// Max length of highlight title in stepper
export const MAX_HIGHLIGHT_TITLE_LENGTH = 60;

// Max highlight sets per enteprise curation
export const MAX_HIGHLIGHT_SETS_PER_ENTERPRISE_CURATION = 8;

// Max number of content items per highlight set
export const MAX_CONTENT_ITEMS_PER_HIGHLIGHT_SET = 12;

// Max number of content items displayed from search results
export const MAX_PAGE_SIZE = 24;

// Stepper Step Labels
export const STEPPER_STEP_LABELS = {
  CREATE_TITLE: 'Create a title',
  SELECT_CONTENT: 'Select content',
  CONFIRM_PUBLISH: 'Confirm and publish',
};

// Stepper Step Text that match testing components
export const STEPPER_STEP_TEXT = {
  HEADER_TEXT: {
    createTitle: 'Create a title for your highlight',
    selectContent: 'Add content to your highlight',
    confirmContent: 'Confirm your selections',
  },
  SUB_TEXT: {
    createTitle: `Create a unique title for your highlight. This title is visible
                  to your learners and helps them navigate to relevant content.`,
    selectContent: (highlightTitle) => sanitizeAndParseHTML(`Select up to <strong>${MAX_CONTENT_ITEMS_PER_HIGHLIGHT_SET}</strong> items for "${highlightTitle}".
                                              Courses in learners&apos; portal appear in the order of selection.`),
    confirmContent: (highlightTitle) => `Review content selections for "${highlightTitle}"`,
  },
  PRO_TIP_TEXT: {
    createTitle: `Pro tip: we recommend naming your highlight collection to reflect skills
              it aims to develop, or to draw the attention of specific groups it targets.
              For example, "Recommended for Marketing" or "Develop Leadership Skills".`,
    selectContent: `Pro tip: a highlight can include courses similar to each other for your learners to choose from,
                    or courses that vary in subtopics to help your learners master a larger topic`,
  },
  ALERT_MODAL_TEXT: {
    title: 'Lose Progress?',
    content: 'If you exit now, any changes you\'ve made will be lost.',
    buttons: {
      exit: 'Exit',
      cancel: 'Cancel',
    },
  },
};

// Header text extracted into constant to maintain passing test on changes
export const HEADER_TEXT = {
  currentContent: 'Highlights',
  catalogVisibility: 'Catalog Visibility',
  zeroStateHighlights: 'You haven\'t created any highlights yet.',
  SUB_TEXT: {
    catalogVisibility: 'Choose a visibility for your catalog.',
    currentContent: `Create up to ${MAX_HIGHLIGHT_SETS_PER_ENTERPRISE_CURATION} highlights for your learners.`,
    zeroStateHighlights: 'Create and recommend content collections to your learners, enabling them to quickly locate content relevant to them.',
  },
  PRO_TIP_TEXT: {
    catalogVisibility: 'Pro tip: regardless of your choice, learners will be able to see all highlight collections.',
  },
};

// Button text extracted from constant to maintain passing test on changes
export const BUTTON_TEXT = {
  createNewHighlight: 'New',
  catalogVisibility: 'New highlight',
  zeroStateCreateNewHighlight: 'New highlight',
  catalogVisibilityRadio1: 'Learners can view and enroll into all courses in your catalog',
  catalogVisibilityRadio2: 'Learners can only view and enroll into highlighted courses',
};

// Button text for stepper help center button
export const STEPPER_HELP_CENTER_FOOTER_BUTTON_TEXT = 'Help Center: Program Optimization';

// Alert Text extracted from constant to maintain passing test on changes
export const ALERT_TEXT = {
  HEADER_TEXT: {
    catalogVisibility: 'No highlights created',
    catalogVisibilityAPI: 'Catalog visibility not updated',
    currentContent: 'Highlight limit reached',
  },
  SUB_TEXT: {
    catalogVisibility: 'At least one highlight has to be created to make a selection',
    catalogVisibilityAPI: 'Something went wrong when updating your setting. Please try again.',
    currentContent: 'Delete at least one highlight to create a new one.',
  },
  GLOBAL_ALERT_TEXT: {
    message: 'Are you sure? Your data will not be saved.',
  },
  TOAST_TEXT: {
    catalogVisibility: 'Catalog visibility settings updated.',
  },
};

// Default footer values based on API response for ContentHighlightCardItem
export const FOOTER_TEXT_BY_CONTENT_TYPE = {
  course: 'Course',
  program: 'Program',
  learnerpathway: 'Pathway',
};

export const LEARNER_PORTAL_CATALOG_VISIBILITY = {
  ALL_CONTENT: {
    value: 'ALL_CONTENT',
    canOnlyViewHighlightSets: false,
  },
  HIGHLIGHTED_CONTENT: {
    value: 'HIGHLIGHTED_CONTENT',
    canOnlyViewHighlightSets: true,
  },
};

// Empty Content and Error Messages
export const DEFAULT_ERROR_MESSAGE = {
  EMPTY_HIGHLIGHT_SET: 'There is no highlighted content for this highlight collection.',
  // eslint-disable-next-line quotes
  EMPTY_SELECTEDROWIDS: 'You don\'t have any highlighted content selected. Go back to the previous step to select content.',
  EXCEEDS_HIGHLIGHT_TITLE_LENGTH: `Titles may only be ${MAX_HIGHLIGHT_TITLE_LENGTH} characters or less`,
};
