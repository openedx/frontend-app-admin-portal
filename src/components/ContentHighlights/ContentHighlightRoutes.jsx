import React from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { ROUTE_NAMES } from '../EnterpriseApp/data/constants';
import ContentHighlightSet from './ContentHighlightSet';
import ContentHighlightsDashboard from './ContentHighlightsDashboard';

const ContentHighlightRoutes = ({ enterpriseSlug }) => {
  const baseContentHighlightPath = `/${enterpriseSlug}/admin/${ROUTE_NAMES.contentHighlights}`;
  return (
    <>
      <Route
        path={baseContentHighlightPath}
        component={ContentHighlightsDashboard}
        exact
      />
      <Route
        path={`${baseContentHighlightPath}/:highlightSetUUID/`}
        component={ContentHighlightSet}
        exact
      />
    </>
  );
};

const mapStateToProps = state => ({
  enterpriseSlug: state.portalConfiguration.enterpriseSlug,
});

ContentHighlightRoutes.propTypes = {
  enterpriseSlug: PropTypes.string.isRequired,
};

export default connect(mapStateToProps)(ContentHighlightRoutes);
