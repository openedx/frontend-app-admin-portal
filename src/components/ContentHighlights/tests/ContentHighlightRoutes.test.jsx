import { renderWithRouter } from '@edx/frontend-enterprise-utils';
import { TEST_ENTERPRISE_SLUG } from '../../../data/tests/constants';
import { ContentHighlightsContext, testHighlightSet } from '../../../data/tests/ContentHighlightsTestData';
import ContentHighlightRoutes from '../ContentHighlightRoutes';

describe('ContentHighlightRoutes', () => {
  it('renders zero state', () => {
    const { container } = renderWithRouter(
      <ContentHighlightsContext>
        <ContentHighlightRoutes />
      </ContentHighlightsContext>,
      { route: `/${TEST_ENTERPRISE_SLUG}/admin/content-highlights` },
    );
    expect(container).toMatchSnapshot();
  });
  it('renders highlight set', () => {
    const { container } = renderWithRouter(
      <ContentHighlightsContext enterpriseAppContextValues={{
        value: {
          enterpriseCuration: {
            enterpriseCuration: {
              highlightSets: [testHighlightSet],
            },
          },
        },
      }}
      >
        <ContentHighlightRoutes />
      </ContentHighlightsContext>,
      { route: `/${TEST_ENTERPRISE_SLUG}/admin/content-highlights/${testHighlightSet.uuid}` },
    );
    expect(container).toMatchSnapshot();
  });
});
