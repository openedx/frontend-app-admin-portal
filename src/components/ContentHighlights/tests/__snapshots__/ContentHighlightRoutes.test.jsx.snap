// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHighlightRoutes renders highlight set 1`] = `
<div>
  <div
    class="mt-5 container-fluid"
  >
    <div
      class="pgn__action-row"
      data-testid="header-skeleton"
    >
      <h2>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
        </span>
      </h2>
      <span
        class="pgn__action-row-spacer"
      />
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
    <div
      class="pgn__card-grid"
    >
      <div
        class="row"
      >
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          <div
            class="pgn__card pgn__card-light card"
            data-testid="card-item-skeleton"
            tabindex="-1"
          >
            <div
              class="pgn__card-wrapper-image-cap vertical"
              data-testid="image-loader-wrapper"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-image-cap-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 140px;"
                >
                  ‌
                </span>
                <br />
              </span>
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-logo-cap"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 41px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-section mt-2"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-section-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 100px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
            <div
              class="pgn__card-footer vertical"
            >
              <span
                aria-busy="true"
                aria-live="polite"
                class="pgn__card-footer-loader"
              >
                <span
                  class="react-loading-skeleton"
                  style="height: 18px;"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="pgn__card-grid__card-item col-xl-3 col-lg-4 col-md-6 col-12"
        >
          ;
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ContentHighlightRoutes renders zero state 1`] = `
<div>
  <div
    class="my-5 container-fluid"
  >
    <div>
      <nav
        class="mb-4.5 pgn__tabs nav nav-tabs"
        role="tablist"
      >
        <a
          aria-selected="true"
          class="nav-item nav-link active"
          data-rb-event-key="Highlights"
          href="#"
          role="tab"
        >
          Highlights
        </a>
        <a
          aria-selected="false"
          class="nav-item nav-link"
          data-rb-event-key="Catalog Visibility"
          href="#"
          role="tab"
        >
          Catalog Visibility
        </a>
        <a
          class="pgn__tab_invisible pgn__tab_more nav-item nav-link"
          href="#"
          role="tab"
          tabindex="-1"
        >
          <div
            class="pgn__dropdown pgn__dropdown-light dropdown"
            data-testid="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="nav-link dropdown-toggle btn btn-link"
              id="pgn__tab-toggle"
              type="button"
            >
              More...
            </button>
          </div>
        </a>
      </nav>
      <div
        class="tab-content"
      >
        <div
          aria-hidden="false"
          class="fade tab-pane active show"
          role="tabpanel"
        >
          <div
            class="mt-5 row"
          >
            <div
              class="col-xl-5 col-lg-8 col-md-9 col-sm-10 col-12"
            >
              <div
                class="pgn__card pgn__card-light card"
                tabindex="-1"
              >
                <div
                  class="bg-light-400 p-4"
                >
                  <div
                    class="pgn__card-wrapper-image-cap vertical"
                  >
                    <img
                      alt=""
                      class="pgn__card-image-cap"
                      src="icon/mock/path"
                    />
                  </div>
                </div>
                <div
                  class="pgn__card-section text-center align-self-center"
                >
                  <h2
                    class="h3 mb-3"
                  >
                    You haven't created any highlights yet.
                  </h2>
                  <p>
                    Create and recommend content collections to your learners, enabling them to quickly locate content relevant to them.
                  </p>
                </div>
                <div
                  class="pgn__card-footer vertical"
                >
                  <button
                    class="btn btn-primary btn-block"
                    data-testid="zero-state-card-New highlight"
                    type="button"
                  >
                    <span
                      class="pgn__icon btn-icon-before"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        focusable="false"
                        height="24"
                        role="img"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                    New highlight
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="fade tab-pane"
          role="tabpanel"
        >
          <div
            class="mt-4.5"
          >
            <h2
              class="mb-3"
            >
              Catalog Visibility
            </h2>
            <p>
              Choose a visibility for your catalog.
            </p>
            <p>
              <strong>
                Pro tip: regardless of your choice, learners will be able to see all highlight collections.
              </strong>
            </p>
          </div>
          <div
            class="mb-4.5 row"
          >
            <div
              class="col-lg-8 col-md-9 col-sm-10 col-12"
            >
              <div
                class="fade alert-content alert alert-danger show"
                role="alert"
              >
                <span
                  class="pgn__icon alert-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
                <div
                  class="pgn__alert-message-wrapper"
                >
                  <div
                    class="alert-message-content"
                  >
                    <div
                      class="alert-heading h4"
                    >
                      No highlights created
                    </div>
                    <p>
                      At least one highlight has to be created to make a selection
                    </p>
                  </div>
                  <div
                    class="pgn__alert-actions pgn__action-row"
                  >
                    <span
                      class="pgn__action-row-spacer"
                    />
                    <button
                      class="btn btn-primary btn-sm"
                      data-testid="catalog-visibility-alert-New highlight"
                      type="button"
                    >
                      <span
                        class="pgn__icon pgn__icon__sm btn-icon-before"
                      >
                        <svg
                          aria-hidden="true"
                          fill="none"
                          focusable="false"
                          height="24"
                          role="img"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                      New highlight
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="container-fluid"
          >
            <div
              class="pgn__form-group"
            >
              <div
                class="pgn__form-control-set"
                id="form-field1"
                role="radiogroup"
              >
                <div
                  class="d-flex align-items-center position-relative"
                >
                  <span
                    class="pgn__action-row-spacer"
                  />
                  <div
                    class="pgn__form-radio pgn__form-control-disabled"
                  >
                    <input
                      checked=""
                      class="pgn__form-radio-input"
                      data-testid="ALL_CONTENT-form-control-button"
                      disabled=""
                      id="form-field2"
                      name="display-content"
                      type="radio"
                      value="ALL_CONTENT"
                    />
                    <div>
                      <label
                        class="pgn__form-label"
                        for="form-field2"
                      >
                        Learners can view and enroll into all courses in your catalog
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="d-flex align-items-center position-relative"
                >
                  <span
                    class="pgn__action-row-spacer"
                  />
                  <div
                    class="pgn__form-radio pgn__form-control-disabled"
                  >
                    <input
                      class="pgn__form-radio-input"
                      data-testid="HIGHLIGHTED_CONTENT-form-control-button"
                      disabled=""
                      id="form-field3"
                      name="display-content"
                      type="radio"
                      value="HIGHLIGHTED_CONTENT"
                    />
                    <div>
                      <label
                        class="pgn__form-label"
                        for="form-field3"
                      >
                        Learners can only view and enroll into highlighted courses
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="fade tab-pane"
          role="tabpanel"
        />
      </div>
    </div>
  </div>
</div>
`;
